pkg/setting/setting_unified_alerting.go

```diff
@@ -53,5 +53,33 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 		}
 	}
 
+	alerting := iniFile.Section("alerting")
```

Comment:
I think we can convey the meaning of what we're trying to do a bit better here.

```golang
// We want unified alerting settings to take precedence over the legacy ones. We check for defaults on the new setting to know whether we should override or not.
cfg.ReadLegacyAlertingSettingsWithWarning(iniFile)

unifiedAlertingExecuteAlerts := ua.Key("execute_alerts").MustBool(defaultAlertingExecuteAlerts)
if unifiedAlertingExecuteAlerts != defaultAlertingExecuteAlerts {
        cfg.UnifiedAlertingExecuteAlerts = unifiedAlertingExecuteAlerts
}

func (cfg *Cfg) ReadLegacyAlertingSettingsWithWarning(iniFile *ini.File) {
	alerting := iniFile.Section("alerting")
	cfg.UnifiedAlertingExecuteAlerts = alerting.Key("execute_alerts").MustBool(defaultAlertingExecuteAlerts)
	if cfg.UnifiedAlertingExecuteAlerts != defaultAlertingExecuteAlerts {
		cfg.Logger.Warn("using legacy setting of 'execute_alerts' please use the configuration option in the `unified_alerting` section.")
	}
	cfg.UnifiedAlertingEvaluationTimeout = time.Second * time.Duration(alerting.Key("evaluation_timeout_seconds").MustInt64(defaultAlertingEvaluationTimeoutSec))
	cfg.UnifiedAlertingMaxAttempts = alerting.Key("max_attempts").MustInt64(defaultAlertingMaxAttempts)
	cfg.UnifiedAlertingMinInterval = alerting.Key("min_interval_seconds").MustInt64(defaultUnifiedAlertingMinInterval)
}
```

The warning should be repeated for every setting we're trying to set.

Reply:
@gotjosh
Reading this again I don't think that we want to warn users when the legacy option is the default (and made them to change it). In addition to this, these settings are loaded always (even if the unified alerting is disabled), this warning would be confusing in that case.
Instead, I think that we have to warn them when the new option (under unified_alerting) is default and we fallback in applying the legacy one.

Reply:
> if legacy is different from default and unified different from default - prefer legacy and print a warning

why in that case we have to prefer the legacy? if the unified is different from default means that deliberately somebody has changed it.

Reply:
My thinking here is that I want users to explicitly **remove** the legacy setting for the new setting to be active. That way we can guarantee the removal of the whole section and have their configuration not fail. 

Unless we don't error out when you have a setting set that doesn't exist?

Reply:
> Unless we don't error out when you have a setting set that doesn't exist?

we don't error in that case; we just never read it and use it.

Reply:
TIL. I take my words back, let's go for preferring unified in that case then.